### PR TITLE
Add dedicated LocalForecastingModel and renaming some model classes

### PR DESCRIPTION
**Pull request by dennisbader**
### Summary
- introduces a dedicted LocalForecastingModel class: no model inherits directly from ForecastingModel anymore -> we can decouple the logic from now on
- renames the local forecasting model base class like DualCovariatesForecastingModel -> FutureCovariatesLocalForecastingModel: I believe we talked once about the naming confusion for this one, and that it should actually be called FutureCovariates... Please correct me if I'm wrong.
  -  feel free to propose different class names, or whether I should revert some of the renaming

### CODE REVIEW

File: darts/models/forecasting/ensemble_model.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ class EnsembleModel(GlobalForecastingModel):
     """
 
     def __init__(
-        self, models: Union[List[ForecastingModel], List[GlobalForecastingModel]]
+        self, models: Union[List[LocalForecastingModel], List[GlobalForecastingModel]]
```
Comment (hrzn) [inline darts/models/forecasting/ensemble_model.py:32]: That's one evidence that this is cleaner, if we needed one :)

File: darts/models/forecasting/forecasting_model.py [hunk 2]
```diff
@@ -1068,6 +1068,20 @@ def load(path: Union[str, BinaryIO]) -> "ForecastingModel":
         return model
 
 
+class LocalForecastingModel(ForecastingModel, ABC):
+    """The base class for "local" forecasting models, handling only single univariate time series.
+
+    Local Forecasting Models (LFM) are models that can be trained on a single univariate target series only. In Darts,
+    most models in this category tend to be simpler statistical models (such as ETS or FFT). LFMs usually train on
+    the entire target series you supplied when calling :func:`fit()` at once. They can also predict in one go with
```
Comment (hrzn) [inline darts/models/forecasting/forecasting_model.py:1076]:
```suggestion
    the entire target series supplied when calling :func:`fit()` at once. They can also predict in one go with
```

File: darts/models/forecasting/forecasting_model.py [hunk 3]
```diff
@@ -1371,12 +1385,18 @@ def _get_encoders_n(self, n) -> int:
...
-    Among other things, it lets Darts forecasting models wrap around statsmodels models
-    having a `future_covariates` parameter, which corresponds to future-known covariates.
+class FutureCovariatesLocalForecastingModel(LocalForecastingModel, ABC):
+    """The base class for future covariates "local" forecasting models, handling single uni- or multivariate target
+    and optional future covariates time series.
 
-    All implementations have to implement the `_fit()` and `_predict()` methods defined below.
+    Future Covariates Local Forecasting Models (FC-LFM) are models that can be trained on a single uni- or multivariate
+    target and optional future covariates series. In Darts, most models in this category tend to be simpler statistical
+    models (such as ARIMA). FC-LFMs usually train on the entire target and future covariates series you supplied when
```
Comment (hrzn) [inline darts/models/forecasting/forecasting_model.py:1394]:
```suggestion
    models (such as ARIMA). FC-LFMs usually train on the entire target and future covariates series supplied when
```

File: darts/models/forecasting/forecasting_model.py [hunk 4]
```diff
@@ -1535,12 +1555,22 @@ def _predict_wrapper(
...
+):
+    """The base class for transferable future covariates "local" forecasting models, handling single uni- or
+    multivariate target and optional future covariates time series. Additionally, at prediction time, it can be
+    applied to new data unrelated to the original series used for fitting the model.
+
+    Transferable Future Covariates Local Forecasting Models (TFC-LFM) are models that can be trained on a single uni-
+    or multivariate target and optional future covariates series. Additionally, at prediction time, it can be applied
+    to new data unrelated to the original series used for fitting the model. Currently in Darts, all models in this
+    category wrap to statsmodel models such as VARIMA. TFC-LFMs usually train on the entire target and future covariates
+    series you supplied when calling :func:`fit()` at once. They can also predict in one go with :func:`predict()`
```
Comment (hrzn) [inline darts/models/forecasting/forecasting_model.py:1569]:
```suggestion
    series supplied when calling :func:`fit()` at once. They can also predict in one go with :func:`predict()`
```

File: darts/models/forecasting/regression_ensemble_model.py [hunk 5]
```diff
@@ -22,7 +19,7 @@
 class RegressionEnsembleModel(EnsembleModel):
     def __init__(
         self,
-        forecasting_models: Union[List[ForecastingModel], List[GlobalForecastingModel]],
+        forecasting_models: List[ForecastingModel],
```
Comment (hrzn) [inline darts/models/forecasting/regression_ensemble_model.py:22]: Note: that's not exactly the same thing as `Union[List[LocalForecastingModel], List[GlobalForecastingModel]]`

**Review by hrzn — APPROVED** [review]
Looks good, I only added a few tiny comments/suggestions.
Thanks for taking the time to make this cleaner.

File: darts/models/forecasting/regression_ensemble_model.py (hunk 5 above)
Comment (dennisbader) [reply darts/models/forecasting/regression_ensemble_model.py:22]: Oh yeah, good catch!

**Pull request merged**